### ISSUE TITLE
[type-annotation-spacing] Not recognizing the `new` keyword

### ISSUE THREAD

**Issue by JimiC** [typescript-eslint/typescript-eslint#1674]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/type-annotation-spacing": "error"
  }
}
```

```TS
type Class = new (...args: any[]) => any;
```

**Expected Result**
Linting passes

**Actual Result**
Linting fails with ` error  Unexpected space before the '=>'  @typescript-eslint/type-annotation-spacing`

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.22.0` |
| `@typescript-eslint/parser`        | `2.22.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `10.18.1` |
| `npm`                              | `6.13.4` |

**Comment by splincode** [comment]
I have same problem

**Comment by jarretmoses** [comment]
My issue was just closed for this I must have missed it in my search but apparently it is resolved https://github.com/typescript-eslint/typescript-eslint/issues/1663

**Comment by JimiC** [comment]
Thanks for letting us know.
